servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java

```diff
@@ -793,9 +794,12 @@ protected Set<Check> checksForEntry(KeyEntry entry) {
               continue;
             }
 
+            Content c = key.getContent();
             EntriesResponse.Entry entry =
-                EntriesResponse.Entry.entry(
-                    fromKey(key.getKey()), key.getType(), key.getContentId());
+                c != null
+                    ? EntriesResponse.Entry.entry(fromKey(key.getKey()), key.getType(), c)
+                    : EntriesResponse.Entry.entry(
+                        fromKey(key.getKey()), key.getType(), key.getContentId());
```

Comment:
nit: would it not be more straight-forward to use an `Entry` builder in this case to avoid conditional operators?